File: content-solutions/dp/fruit-pyramid.js
function solution(pyramid) {
  const best = [...pyramid[pyramid.length - 1]];
  for (let r = pyramid.length - 2; r >= 0; r--) {
    for (let c = 0; c <= r; c++) best[c] = pyramid[r][c] + Math.max(best[c], best[c + 1]);
  }
  return best[0];
}

File: content-solutions/dp/fruit-pyramid.java
class Solution {
    public int solution(int[][] pyramid) {
        int[] best = pyramid[pyramid.length - 1].clone();
        for (int r = pyramid.length - 2; r >= 0; r--) {
            for (int c = 0; c <= r; c++) best[c] = pyramid[r][c] + Math.max(best[c], best[c + 1]);
        }
        return best[0];
    }
}
